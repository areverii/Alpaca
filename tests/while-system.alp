component Name:
    string name
end 

component Stats:
    int age
    float height
end 

component Color: 
    string color 
end

entity Person:
    Stats
    Name
end    

entity Circle: 
    Color 
end

function void main(<string> args):
    spawn Person:
        Stats:
            age: 22
            height: 5.8
        end

        Name:
            name: "Phila"
        end
    end

    spawn Circle:
        Color:
            color: "RED"
        end
    end

    spawn Circle:
            Color:
                color: "blue"
            end
        end

    spawn Circle:
            Color:
                color: "purple"
            end
        end

    spawn Person:
        Stats:
            age: 21
            height: 6.5
        end

        Name:
            name: "Nick"
        end
    end

    run()
end

system run():
    query<a> [Stats, Name] as people
    query<b> [Color] as colored
    entity<a> person
    entity<b> color
    int i

    for person in people:
        printi(person.Stats.age)
        printf(person.Stats.height)
        prints(person.Name.name)
    end
    
    for i in range(0, 2):
        printi(i)
    end

end
