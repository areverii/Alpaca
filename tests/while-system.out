22
5.8
Phila
21
6.5
Nick
0
1
